framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -254,6 +255,46 @@ public void insertValue(FlowExpressions.Receiver r, AnnotationMirror a) {
         insertValue(r, analysis.createSingleAnnotationValue(a, r.getType()));
     }
 
+    /**
+     * Add the annotation {@code a} for the expression {@code r} (correctly deciding where to store
+     * the information depending on the type of the expression {@code r}).
+     *
+     * <p>This method does not take care of removing other information that might be influenced by
+     * changes to certain parts of the state.
+     *
+     * <p>If there is already a value {@code v} present for {@code r}, then the greatest lower bound
```

Comment:
Is this only safe if the method is also pure? If the method is non-pure, it could break any other property and the only thing we can rely on is the postcondition from this method, nothing that held before.

Reply:
If the method isn't side-effect-free, then the values that might have been changed by the method call are removed from the store before postcondition values are added.  I'll add a test for this.

Reply:
I'll add a note, too.

Reply:
I added this comment to the callsite in CFAbstractTransfer.